Makefile

```diff
@@ -0,0 +1,34 @@
+# SPDX-FileCopyrightText: 2022 Risk.Ident GmbH <contact@riskident.com>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+.PHONY: deps
```

Comment:
The project currently uses an Earthfile for build jobs. We may want to consider using it for other jobs as well, to not have our jobs spread across several files. Other advantages are portability of docker (and less .PHONY).

Reply:
It's so slow though :sweat_smile: I see the benefit of using Earthly to create the artifacts, which isn't a common task, and requires good reproducability. But just for running linters it's kind of too heavy. Also it doesn't support the dependency stuff, like how `make node_modules` will skip if `package.json` is older than `node_modules`.

And the fact that Earthly tries to pull the base images on every build makes it slow. For example `earthly +deps` takes 7 seconds, and that's when the layers are cached. There's an open ticket about this (<https://github.com/earthly/earthly/issues/879>), but no activity for more than a year.

This is a broader discussion that we should have. I'll bring it up tomorrow during daily so we can discuss it verbally, which is usually easier.

Reply:
Yea you've mentioned just before as well, and I've looked at it, but I'm not that charmed by it. It's basically a glorified `alias` / bash script file.

`make` isn't perfect, but `just` isn't adding anything to the party. You know, at some point I'm probably going to [make my own build system](https://xkcd.com/927/) with all the features I want, like:

- Tool dependency management like earthly, to define "I want python image with `pip install reuse`", but with good caching. But no weird "buildkit inside a container" stuff. Just a simple `docker build`, thankyouverymuch.
- File dependencies like `make` to skip rerunning stuff
- Support its own state file where a target so it can skip rerunning when the outcome isn't a binary
- Support running in containers like earthly, but also have volumes to persist stuff like the `~/go/` directory for Go tasks, or `node_modules` for NodeJS stuff.

But alas, that's for another day.

Reply:
I don't really see how avoiding rerunning when a target file has been created can be considered an advantage, when we have to explicitly opt out of that behaviour literally for every single task? 😅  And we are using it exactly as, in your words, a glorified `alias` / bash script file. The whole concept might sometimes be useful for using it as a build tool, but even then wanting rebuilds or not usually is more involved than just "a file already exists" because that does not handle sources that have changed. It is basically a crutch for not having a proper dependency management tool.
Not pretending that `just` is doing anything crazy special, it really isn't. it's just less terrible than make.
Honestly looking forward to your dream tool ;D

Reply:
mind changed: @jilleJr  showed me some nice use cases for file dependencies. I'm okay with Makefile as a pragmatic solution, we can resolve this.